lib/settings/pages/user_labels_settings_page.dart

```diff
@@ -0,0 +1,203 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:smooth_highlight/smooth_highlight.dart';
+import 'package:thunder/account/models/user_label.dart';
+
+import 'package:thunder/core/enums/local_settings.dart';
+import 'package:thunder/post/utils/user_label_utils.dart';
+import 'package:thunder/shared/dialogs.dart';
+import 'package:thunder/shared/full_name_widgets.dart';
+import 'package:thunder/shared/input_dialogs.dart';
+
+class UserLabelSettingsPage extends StatefulWidget {
+  final LocalSettings? settingToHighlight;
+
+  const UserLabelSettingsPage({super.key, this.settingToHighlight});
+
+  @override
+  State<UserLabelSettingsPage> createState() => _UserLabelSettingsPageState();
+}
+
+class _UserLabelSettingsPageState extends State<UserLabelSettingsPage> with SingleTickerProviderStateMixin {
+  GlobalKey settingToHighlightKey = GlobalKey();
+  LocalSettings? settingToHighlight;
+
+  List<UserLabel> userLabels = [];
+
+  void _updateChangedUserLabel(({UserLabel? userLabel, bool deleted}) result) {
+    if (result.userLabel != null) {
+      UserLabel? existingLabel = userLabels.firstWhereOrNull((userLabel) => userLabel.username == result.userLabel!.username);
+      if (existingLabel == null && !result.deleted) {
+        // It doesn't exist in our list yet, add it!
+        setState(() => userLabels.add(result.userLabel!));
+      } else if (existingLabel != null) {
+        if (result.deleted) {
+          // It exists in our list and was deleted, so remove it.
+          setState(() => userLabels.removeWhere((userLabel) => userLabel.username == result.userLabel!.username));
+        } else {
+          // It exists in our list but was changed, so update it.
+          setState(() => userLabels[userLabels.indexOf(existingLabel)] = result.userLabel!);
+        }
```

Comment:
This is more of a preference thing (so no need to change anything!), but I generally like to return early on these types of boolean conditions to reduce the logic. Of course, this only works if this is the last thing that needs to be performed.

```suggestion
        if (result.deleted) {
          // It exists in our list and was deleted, so remove it.
          setState(() => userLabels.removeWhere((userLabel) => userLabel.username == result.userLabel!.username));
          return;
        }
        
        // It exists in our list but was changed, so update it.
        setState(() => userLabels[userLabels.indexOf(existingLabel)] = result.userLabel!);
```

Reply:
That makes sense! In this case, since there's only one line of logic in the `else`, I think the parallelism of having both statements in conditions helps to see what's going on without too much clutter.